src/data.rs

```diff
@@ -120,7 +120,14 @@ impl IndexMut<usize> for Value {
 impl IndexMut<&str> for Value {
     fn index_mut(&mut self, i: &str) -> &mut Value {
         match self {
-            Value::Object(o) => &mut o[i],
+            Value::Object(o) => {
+                if o.contains_key(i) {
+                    &mut o[i]
+                } else {
+                    o.insert(i.to_string(), Value::Null);
+                    &mut o[i]
```

Comment:
This bit of magic was necessary to allow arbitrary adding of elements to a data dict